### DISCUSSION TITLE
¿Cómo creamos la dinámica para el trabajo en equipo (tribus) durante el semestre?

### DISCUSSION THREAD

**Discussion by tapilew** [hackatomik/SISCOx256#7] [Planear Curso | Lab]
# Propuesta de Dinámica de Trabajo en Tribus para el Curso de Sistemas Colaborativos

## **Objetivo General**

Fomentar el trabajo en equipo a través de la creación de **tribus** que tendrán la opción de desarrollar un proyecto como una **"startup"** o como una **"fábrica de software"**. Esta dinámica permitirá a los estudiantes aplicar metodologías ágiles, colaborar en entornos de desarrollo reales, y presentar su trabajo en un evento final donde se evaluará tanto el producto desarrollado como la efectividad del equipo.

### **1. Opción: Startup**
- **Objetivo:** Desarrollar un software innovador desde cero, asumiendo el rol de una startup tecnológica.
- **Repositorios:** Cada tribu gestionará su propio repositorio de GitHub, donde desarrollarán su proyecto, documentarán el progreso y manejarán issues.
- **Metodología:** Se utilizará Scrum, con Sprints de 2-3 semanas para iterar sobre el producto.
- **Entrega Final:**
  - **Pitch:** Presentarán un pitch titulado "¿Por qué debería recibir inversión para mejorar el software que he desarrollado?".
  - **Contenido del Pitch:** Justificación de la viabilidad del software, impacto potencial en el mercado, y cómo una inversión podría escalar el proyecto.
  - **Documentación:** Entregarán toda la documentación relacionada con el desarrollo del software, incluyendo análisis de mercado, modelo de negocio, y un roadmap para mejoras futuras.

### **2. Opción: Fábrica de Software**
- **Objetivo:** Colaborar en proyectos liderados por el profesor Víctor o Hackatomik [H9], simulando un entorno de "fábrica de software".
- **Repositorios:** Los miembros trabajarán en repositorios compartidos ya existentes, contribuyendo al desarrollo de proyectos más grandes.
- **Metodología:** Aplicarán prácticas de desarrollo colaborativo, control de versiones, revisión de código, integración continua, y Scrum para la gestión del equipo.
- **Entrega Final:**
  - **Pitch:** Presentarán un pitch titulado "¿Por qué somos un equipo de software efectivo?".
  - **Contenido del Pitch:** Argumentar cómo la tribu ha contribuido al éxito del proyecto, demostrando la calidad del código, eficiencia en la entrega, y capacidad de colaborar efectivamente.
  - **Documentación:** Incluir documentación técnica sobre las contribuciones realizadas, métricas de productividad, y casos de éxito del software desarrollado.

### **1. Formación de Tribus y Selección de Opción**
- Tras la formación de las tribus, cada equipo discutirá internamente para decidir si desean actuar como una "startup" o como una "fábrica de software".
- Una vez tomada la decisión, cada tribu deberá comunicar su elección al profesor y recibir la orientación específica para su tipo de proyecto.

### **2. Desarrollo del Proyecto**
- **Metodología:** Ambas opciones seguirán metodologías ágiles (Scrum), con Sprints de 2-3 semanas para planificar, desarrollar, y revisar su progreso.
- **Evaluaciones:** Se realizarán evaluaciones periódicas basadas en los avances reportados en los repositorios de GitHub y en las reuniones de Sprint.

### **3. Preparación del Pitch Final**
- **Ensayos:** Cada tribu realizará ensayos de su pitch final para recibir retroalimentación y perfeccionar su presentación.
- **Revisión de Documentación:** Asegurar que toda la documentación esté completa y lista para ser entregada junto con el pitch.

### **4. Presentación Final**
- **Evento:** Organizar un evento de presentación en el lobby de FISC, donde cada tribu expondrá su pitch ante un jurado compuesto por profesores y profesionales del sector.
- **Evaluación:** El jurado evaluará tanto el contenido del pitch como la calidad del software desarrollado, teniendo en cuenta la documentación entregada y la efectividad del equipo.

## **Encuestas Pre-Implementación**

Para optimizar la formación de tribus y la asignación de roles, se administrarán encuestas pre-implementación que capturen las expectativas, habilidades y áreas de interés de los estudiantes. Estos datos serán clave para formar equipos balanceados y asegurar que cada estudiante esté en una tribu donde pueda contribuir y aprender de manera significativa.

### **Encuesta Pre-Implementación**
- **Objetivo:** Capturar expectativas, habilidades, y preferencias de roles de los estudiantes.
- **Administración:** Las encuestas se administrarán antes de la formación de tribus.
- **Resultados:** Los resultados guiarán la asignación de miembros y roles en cada tribu.

**Comment by tapilew** [comment]
Estamos abiertos a comentarios sobre la propuesta. Y por supuesto, que hagan sus propias propuestas para diseñar el mejor curso posible juntos :)

Nada está escrito en piedra así que comenten todo lo que quieran.

**Comment by quantumquirkxyz** [comment]
Para crear una dinámica efectiva de trabajo en equipo durante el semestre, podemos seguir un enfoque estructurado y adaptado a las necesidades del curso. Primero, formaremos tribus de estudiantes y cada grupo elegirá entre dos opciones de proyecto: desarrollar una startup o colaborar en una fábrica de software. En el primer caso, cada tribu actuará como una startup tecnológica, gestionando su propio repositorio de GitHub y utilizando metodologías ágiles como Scrum para desarrollar un software innovador. En el segundo caso, las tribus trabajarán en proyectos existentes liderados por el profesor o Hackatomik [H9], aplicando prácticas de desarrollo colaborativo y control de versiones.

El desarrollo del proyecto se basará en sprints de 2-3 semanas, con evaluaciones periódicas del progreso y reuniones de sprint. Al final del semestre, cada tribu presentará un pitch explicando su proyecto y su eficacia como equipo. Además, se organizará un evento de presentación donde se evaluará el trabajo en base al pitch y la calidad del software. Antes de formar las tribus, se administrarán encuestas para entender las expectativas y habilidades de los estudiantes, asegurando que cada grupo esté equilibrado y pueda contribuir de manera efectiva al proyecto.
